### PR TITLE
Multiple merger

### PR DESCRIPTION
**Pull request by JereKoskela**
- Replaced references to Ne with N in the multiple merger documentation.
- Changed `truncation_point` for the Beta-coalescent as discussed in #1216. In particular, the range is now (0, inf] and the default is inf, which gives the same behaviour as the previous default. Some tests had to change too because they were testing for the validity of the previous (0,1] range, and were now expecting errors from valid parameters, or vice versa.
- Fixed a minor error in the mean juvenile count `m` under the Beta-coalescent, and simplified some computations by extracting out two new static functions to perform some brief computations.

Ping @eldonb - I can't request you as a reviewer for some reason.

### CODE REVIEW

**Comment by JereKoskela** [comment]
The lint / pre-commit complaint is interesting: there is a `numpy.inf` hidden in `obj_copy = eval(repr(obj), globals(), msprime.__dict__)` on line 1785 of tests/test_ancestry.py when `model = msprime.BetaCoalescent` with `truncation_point = inf`. If I don't tell it what `numpy.inf` is, then the test fails, but if I do, then pre-commit complains. Any suggestions?

File: lib/msprime.c
Comment (jeromekelleher) [inline lib/msprime.c:6375]: If we set the input value to ``DBL_MAX`` then I think this condition isn't necessary. I think (but we'd need to verify, just t to be sure) that ``BDL_MAX / (DBL_MAX  + m)`` is 1 for any reasonable value of m.

The way to specify DBL_MAX in python is via ``sys.float_info.max``.

**Comment by TPPSellinger** [comment]
I don't think I can help you with the  numpy.inf but I will check your modifications  in the c code and all the tests in verifications.py.

File: lib/msprime.c
Comment (jeromekelleher) [inline lib/msprime.c:6849]: I think we should put a ``isfinite(truncation_point)`` check here as well. Dealing with floating point special classes is a PITA, it's much easier if we can use DBL_MAX all the way through.

**Review by jeromekelleher — COMMENTED** [review]
LGTM @JereKoskela. I think we should change from ``inf`` to DBL_MAX though, and hopefully this gets rid of the special case in the computation.

File: lib/msprime.c
Comment (JereKoskela) [reply lib/msprime.c:6375]: Is there much harm in keeping it as is? It costs a few extra lines of code and an if-statement, but the value of `m` diverges as alpha approaches 1, so in principle `DBL_MAX / (DBL_MAX + m) == 1` should fail for sufficiently small but otherwise valid values of alpha.

Comment (jeromekelleher) [reply lib/msprime.c:6375]: Yeah, OK, it's probably safer to have the special case and I guess it all works either way.

File: lib/msprime.c
Comment (jeromekelleher) [reply lib/msprime.c:6849]: Although, I guess it's more mathematically satisfying to say ``inf`` rather than "arbitrary very big number". We'll have to sort out the annoying provenance stuff if we use ``inf``. I can put in a commit to do that if you prefer being able to use ``np.inf`` e.g. as the input value.

Comment (JereKoskela) [reply lib/msprime.c:6849]: I used the first thing I could make work. I'm still not very familiar with Python, and am completely unfamiliar with the C-Python interface. I'm happy to swap to `sys.foat_info.max` from above, and add the `isfinite` check.

**Comment by jeromekelleher** [comment]
Looks good to me, thanks @JereKoskela. Good to go after a squash and we get @TPPSellinger's approval I think.

**Comment by TPPSellinger** [comment]
Sorry I'm late. I went through all modifications and I didn't find anything. I also ran all beta related test in verification.py. Everything looks fine.

**Review by jeromekelleher — APPROVED** [review]

**Comment by jeromekelleher** [comment]
Thanks very much @JereKoskela, this is starting to look very polished. 

ps. One tiny thing: If you add a line "Closes #1216" to the commit message GitHub will take care of the issue closing. I've linked the issue here manually, which will hopefully do the same thing.

**Pull request merged**
